src/common/Debugging.h

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright (c) 2018 RISC Software GmbH
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+#include <map>
+
+class TopoDS_Shape;
+
+namespace tigl
+{
+void dumpShape(const TopoDS_Shape& shape, const std::string& outputDir, const std::string& filename, int counter = -1);
+
+// represents a point in the execution of a program at which shapes can be dumped
+class TracePoint
+{
+public:
+    TracePoint(const std::string& outputDir);
+
+    void operator++(int);
```

Comment:
Interesting! When is this operator actually called? I did not see any code for this.

Reply:
It's called from code expanded from the TRACE_POINT_OUTPUT_DIR macro